control

```diff
@@ -487,7 +490,7 @@ class Step(collections.Callable):
         self.context = context
         if advance:
             self.context.index += 1
-        self.tag = str(self.context.index)
+        self.tag = "%d-%s" % (str(self.context.index), MYHOSTNAME)
```

Comment:
I suspect you might not have actually tested this... (hint: %d does not go well with str())